eventer.go

```diff
@@ -1,36 +1,116 @@
 package gobot
 
+type eventChannel chan *Event
+
 type eventer struct {
-	events map[string]*Event
+	// map of valid Event names
+	eventnames map[string]string
+
+	// new events get put in to the event channel
+	in eventChannel
+
+	// slice of out channels used by subscribers
+	outs []eventChannel
 }
 
-// Eventer is the interface which describes behaviour for a Driver or Adaptor
-// which uses events.
+// Eventer is the interface which describes how a Driver or Adaptor
+// handles events.
 type Eventer interface {
-	// Events returns the Event map.
-	Events() (events map[string]*Event)
-	// Event returns an Event by name. Returns nil if the Event is not found.
-	Event(name string) (event *Event)
-	// AddEvent adds a new Event given a name.
+	// Events returns the map of valid Event names.
+	Events() (eventnames map[string]string)
+	// Event returns the map of valid Event names.
```

Comment:
Doc here seems incorrect, since `Event` returns a string

Also wondering if maybe `Event` shouldn't be `GetEvent` or something to match up as an obvious companion with `AddEvent`.